File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        int possibleAns = 0;
        for(int i=0;i<k;i++){
            possibleAns += cardPoints[i];
        }
        int ptr = k-1,ans = possibleAns;
        for(int i=n-1;i>=(n-k);i--){
            possibleAns -= cardPoints[ptr];
            ptr--;
            possibleAns += cardPoints[i];
            ans = max(ans,possibleAns);
        }
        return ans;
    }
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.go
func maxScore(cardPoints []int, k int) int {
    n := len(cardPoints)
    possibleAns := 0
    for i:=0;i<k;i++ {
        possibleAns += cardPoints[i]
    }
    ans := possibleAns
    ptr := k-1
    for i:=n-1;i>=(n-k);i--{
        possibleAns -= cardPoints[ptr];
        ptr--;
        possibleAns += cardPoints[i]
        if possibleAns > ans {
            ans = possibleAns
        }
    }
    return ans
}